plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceProviderDescriptor.java

```diff
@@ -367,6 +370,9 @@ public boolean removeDriver(DriverDescriptor driver)
     // Native clients
 
     public List<NativeClientDescriptor> getNativeClients() {
+        if (inheritClients && parentProvider != null) {
+            nativeClients.addAll(parentProvider.getNativeClients());
+        }
```

Comment:
It will bloat `nativeClients` every time this method is called.

Can we populate it in the constructor, just after reading the configuration?